app/Http/Controllers/Admin/PostController.php

```diff
@@ -0,0 +1,202 @@
+
+            return '<img src="' . $url . '" alt="" class="anyway_image">';
+        })
+        ->rawColumns(['action', 'name_parent', 'image'])
+        ->toJson();
+    }
+
+    public function store(StorePost $request)
+    {
+        $data = $request->all();   
+        $data['lang_parent_id'] = 0;
+        $lang = $this->langRepository->whereFirst('short', config('language.short'));
+        if(is_null($lang)) {
+            return response()->json(['error' => __('messages.Notfound')]);;
+        }
+        $data['lang_id'] = $lang['id'];
+        if ($request->hasFile('file')) {
+            $data['image'] = uploadImage(config('upload.default'), $request->file);
+        }
+        DB::beginTransaction();
+        try {
+            $post = $this->postRepository->create($data);
+            $langMap = $post['id'];
+            $this->postRepository->update($post['id'], ['lang_map' => $langMap]);
+            DB::commit();
+        } catch (Exception $e) {
+            DB::rollBack();
+            throw new Exception($e->getMessage());
+        }
+
+        return response()->json(['success' => __('messages.Add Successfully'), 'error' => false]);
+    }
+
+    public function edit($id)
+    {
+        $post = $this->postRepository->find($id);
+        if (is_null($post)) {            
+            return response()->json(['error' => __('messages.Notfound')]);
+        }
+        $category = $this->cateRepository->find($post['cate_id']);
+        if(is_null($category)) {
+            return response()->json(['error' => __('messages.Notfound')]);
+        }
+        $post['cate_name'] = $category['name'];
+        $post['image'] = asset(config('upload.default')) . '/' . $post['image'];
+
+        return $post;
+    }
+
+    public function update(UpdatePost $request)
+    {
+        $data = $request->all();
+        $post = $this->postRepository->find($data['post_id']);
+        if (is_null($post)) {
+            return response()->json(['error' => __('messages.Notfound')]);;
+        }
+        if ($request->hasFile('file')) {
+            $data['image'] = uploadImage(config('upload.default'), $request->file);
+        } else {
+            $data['image'] = $post['image'];
+        }
+        $posts = $this->postRepository->whereall('lang_map', $post['lang_map']);
+        $post_id = $posts->pluck('id')->toArray();
+        DB::beginTransaction();
+        try {
+            $array = array('image' => $data['image']);
+            Post::whereIn('id', $post_id)->update($array);
+            $this->postRepository->update($data['post_id'], $data);
+            DB::commit();
+        } catch (Exception $e) {
+            DB::rollBack();
+            throw new Exception($e->getMessage());
+        }
+
+        return response()->json(['success' => __('messages.Update Successfully'), 'error' => false]);
+    }
+
+    public function trans(StorePost $request) {
+        $data = $request->all();
```

Comment:
các function store này viết vào scope hoặc repository chứ?

Reply:
function này là translate ngôn ngữ c ạ
Em phải xử lí mấy bước mới lưu vào vào,
tông nó dài dài thế thôi chứ em toàn kiểm tra nó xem có rỗng hay k, nếu rỗng return notfound ạ